src/handlers/mxcontrol.js

```diff
@@ -0,0 +1,159 @@
+const nconf = require("nconf");
+const MXControlEvent = require("../events/mxcontrol");
+const Hipchat = require("../api/hipchat");
+const MXControl = require("mxcontrol");
+const Promise = require("bluebird");
+const dnsLookup = Promise.promisify(require("dns").lookup);
+const {mxDynamoDB} = require("mxaws");
+
+nconf.env("_");
+
+const appNames = ["MXWeb", "MXBackend"];
+const hipchat = new Hipchat(nconf.get("HIPCHAT:ROOM:MXCONTROL:TOKEN"));
+const dynamoName = nconf.get("DYNAMODB:TABLE:MXCONTROL");
+const statusVerbs = [...MXControl.possibleActions.statusVerbs];
+const upVerbs = [...MXControl.possibleActions.upVerbs];
+const rebootVerbs = [...MXControl.possibleActions.rebootVerbs];
+
+const handleMXControlEvent = async event => {
+    const useFullCLI = event.debug;
+    const task = event.task;
+    const targetName = task.instance || task.environment || task.database;
+    const action = task.action.toLowerCase();
+
+    if (!checkForControlTaskErrors) return;
+
+    msgMXControlRoom(
+        MXControl.buildLog(targetName, task.action, task.size, task.database));
+
+    //Standard MXControl status functions weren't meant for a hipchat audience.
+    //Wrap those with something new.
+    if (statusVerbs.includes(action)) {
+
+        const statusResponse = await MXControl.runTask(task);
+
+        if (useFullCLI) //If using the full CLI anyway, format output for Hipchat
+            await msgMXControlRoom(formatStatusResponse(statusResponse).join("\n"));
+
+        else if (!useFullCLI && targetName) {
+
+            const environmentErrors =
+                await checkEnvironmentReadiness(statusResponse, targetName);
+
+            if (environmentErrors.length != 0) {
+                await msgMXControlRoom(`${targetName}.massexchange.com is **NOT READY.**`);
+                environmentErrors.map(async err => await msgMXControlRoom(err));
+
+            } else await msgMXControlRoom(`${targetName}.massexchange.com is **READY.**`);
+
+        } else await msgMXControlRoom("Please specify an environment");
+
+        return;
+    }
+
+    if (!useFullCLI &&
+        (upVerbs.includes(action) || rebootVerbs.includes(action)))
+        addEnvEntriesToDynamo(targetName);
+
+    MXControl.runTask(task).catch(err =>{
```

Comment:
I don't want to await here. I want to fire and forget -- waiting is worthless in Lambda, remember?

Reply:
it will still work the exact same way, there will be no effective difference. its a matter of syntax. since you're already using the `async/await` syntax in the rest of the function, i suggested this change for consistency's sake

Reply:
It isn't correct, though. We don't want to stall and keep lambda operating. Shit costs money(albeit pennies,) let's not be wasteful.